python/hail/dataset.py

```diff
@@ -1406,13 +1421,65 @@ def export_variants_solr(self, variant_condition, genotype_condition,
         self.hc._run_command(self, pargs)
 
     def export_vcf(self, output, append_to_header=None, export_pp=False, parallel=False):
-        """Export as .vcf file.
+        """Export a VariantDataset as a .vcf or .vcf.bgz file.
+
+        **Examples**
+
+        Export to VCF as an uncompressed file:
+
+        >>> hc.read('data/example.vds').export_vcf('data/example.vcf')
+
+        Export to VCF as block-compressed files with linear-scaled dosages:
+
+        >>> hc.read('data/example.vds').export_vcf('data.example.vcf.bgz', export_pp=True, parallel=True)
+
+        Export a sites-only VCF with addition header lines from `myheader.txt`:
+
+        >>> (hc.read('data/example.vds')
+        >>>    .filter_samples_all()
+        >>>    .export_vcf('data/sites-only.vcf.bgz', append_to_header='data/myheader.txt')
+
+        **Notes**
+        py:meth:`exportvcf` exports the VDS the VCF format as described in the [VCF 4.2 spec](https://samtools.github.io/hts-specs/VCFv4.2.pdf).
+
+        - Use the ``.vcf.bgz`` extension in the output file name for `blocked GZIP <http://www.htslib.org/doc/tabix.html>`_ compression. We strongly recommended this for exporting large VCFs.
+        - Set ``parallel=True`` to return a set of VCF files (one per partition) rather than serially concatenating these files.
+        - Set ``append_to_header=file.txt`` to append *file.txt* to the output VCF header.
+        - Set ``export_pp=True`` to export linear-scaled probabilities (Hail's pp field on genotype) as the VCF PP FORMAT field.
+
+        Importing a VCF to a Hail VDS and then exporting to a VCF will not result in exactly the same VCF.  The differences, however, are limited to the header and INFO field.  The output VCF header will contain:
+
+        - FORMAT lines, with the format "GT:AD:DP:GQ:PL", or "GT:AD:DP:GQ:PP" if ``export_pp=True``
+        - FILTER lines, if present in the imported VCF
+        - INFO lines (see exported annotations below)
+
```

Comment:
It would be nice if you could format this section better. Super difficult to read.